scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
Do you mean `safe_url_string` called twice for Unicode input in Python 2 case?
I believe with `LinkExtractor` it won't happen with [this change](https://github.com/scrapy/scrapy/blob/link-encoding/scrapy/linkextractors/lxmlhtml.py#L60) from this very PR (`safe_url_string` returns `str` in Python2)

Sorry, I don't understand the relation to #833 . That PR is about `Request`, not `Link`.
Am I missing something?